docs/okeps/okep-5377-extend-udn-to-support-multiple-cluster-subnets-in-layer3-topology.md

```diff
@@ -0,0 +1,284 @@
+  incrementally over time without recreating the network.
+- **Pod density variance:** Nodes with different resource capacities may need subnets of different
+  sizes. Supporting multiple subnets enables better IP utilization by allocating appropriately sized
+  subnets per node (based on node hints).
+  _Note: Using different `hostSubnet` sizes is out of scope for this OKEP._
+
+## Goals
+
+* Support multiple cluster subnets per IP family in UDN's Layer3 topology.
+* Align UDN’s subnet capabilities with existing support in NAD.
+* Allow dynamic subnet expansion by appending new subnets to an existing UDN.
+
+## Non-Goals
+
+* Supporting multiple subnets in Layer2 or Localnet topologies.
+* Supporting deletion of existing subnets or modification of their `hostSubnet` size.
+
+## Future-Goals
+
+* Support subnets with different `hostSubnet` sizes.
+  When the maximum number of Pods on a node is limited by hardware resources (e.g., GPUs), the subnet
+  allocator can assign the smallest fitting subnet to the node by leveraging a node-specific hint.
+  This allows a single CIDR to serve more nodes.
+  This feature depends on an OVN commit [27cc274](https://github.com/ovn-org/ovn/commit/27cc274e66acd9e0ed13525f9ea2597804107348).
+
+## Introduction
+
+When using UDN to define a network in Layer3 topology, each node gets a subnet allocated from the
+UDN's `subnets` list per IP family. Currently, UDN limits `subnets` to a single subnet per IP
+family, enforced by CRD validation and full immutability of the UDN spec. This limitation blocks
+practical use cases where operators need to expand IP space incrementally — a feature already
+available through NAD.
+
+This proposal aims to bring UDN's flexibility in line with NAD by supporting multiple cluster
+subnets and enabling controlled mutability.
+
+
+## User-Stories/Use-Cases
+
+### Story 1: Expanding subnet pool for growing clusters
+
+**As a** cluster operator,
+**I want** to expand the available IP address pool for a Layer3 network,
+**so that** I can scale the cluster without redeploying or disrupting workloads.
+
+
+## Proposed Solution
+
+### API Details
+
+1. Update `UDN`/`CUDN` CRD to allow `layer3.subnets` to accept multiple subnets per IP family.
+2. Allow updates to the `layer3.subnets` field, but only to **append** new entries.
+   - Reject updates that attempt to remove or modify existing subnet entries.
+   - Reject new entries that overlap with existing subnets or use a different `hostSubnet`.
+
+
+### Implementation Details
+
+### Node Subnet Allocation
+
+The current NAD implementation already supports multiple subnets in Layer3 topology. However,
+when a new subnet is added to a NAD, the corresponding network controller is first deleted and then
+recreated with the updated subnet list. This behavior can cause node subnets to be reallocated,
+resulting in unnecessary disruption.
+
+In practice, this issue can often be worked around by restarting the `ovnkube-cluster-manager`, so
```

Comment:
This section is confusing to me. I'm not sure what you are suggesting here as the implementation details. Certainly requiring restarting ovnkube-cluster-manager is not an option. I would expect that when the UDN/NAD is updated with an additional subnet it is handled gracefully.

Reply:
This section depicts a issue of current implementation. 

Updated this section to eliminate the confusion.